File: DSP1_S3_JAVA/src/CadenaFiltros.java
import java.util.ArrayList;

public class CadenaFiltros {
    ArrayList<Filtro> filtros;

    public CadenaFiltros(){
        filtros = new ArrayList<Filtro>();
    }

    public void addFiltros(Filtro f){
        filtros.add(f);
    }

    public void deleteFiltros(Filtro f){
        System.out.println("El filtro ha sido borrado\n");
        filtros.remove(f);
    }

    public void procesarFormulario(Formulario formulario){
        for (Filtro f : filtros)
            f.procesar(formulario);
    }
}

File: P1/P1S3/CadenaFiltros.cpp
//
// Created by ines on 8/4/21.
//

#include "CadenaFiltros.h"
#include <vector>
#include <algorithm>

CadenaFiltros::CadenaFiltros() {

}

void CadenaFiltros::addFiltros(Filtro* f) {
    filtros.push_back(f);
}

void CadenaFiltros::deleteFiltros(Filtro *f) {
    vector<Filtro*>::iterator it = std::find(std::begin(filtros), end(filtros), f);

    if (it != filtros.end()){
        filtros.erase(it);
        std::cout << "\nEl filtro ha sido borrado" << endl << endl;
    }

}

void CadenaFiltros::procesarFormulario(Formulario &formulario) {
    Filtro *f;

    for(unsigned i = 0; i < filtros.size(); ++i) {
        filtros[i]->procesar(formulario);
    }
}